tests/api2/test_cloud_sync.py

```diff
@@ -218,21 +219,29 @@ def test_state_persist():
 
 
 if ha:
+    def get_controllers_ips():
+        return os.environ.get('controller1_ip'), os.environ.get('controller2_ip')
+
+    def verify_file_contents_match(path1, path2):
+        ctrl1_local_tmp_file = '/tmp/ctrl1.txt'
+        ctrl2_local_tmp_file = '/tmp/ctrl2.txt'
+        ctrl1_ip, ctrl2_ip = get_controllers_ips()
+        assert all((ctrl1_ip, ctrl2_ip)), 'Unable to determine HA controller IPs'
+
+        with get_remote_file_temporarily(path1, ctrl1_local_tmp_file, _ip=ctrl1_ip) as rv1:
+            assert rv1['result'], f'Failed to download {path1!r} from {ctrl1_ip}'
+            with get_remote_file_temporarily(path2, ctrl2_local_tmp_file, _ip=ctrl2_ip) as rv2:
+                assert rv2['result'], f'Failed to download {path2!r} from {ctrl2_ip}'
+                with open(ctrl1_local_tmp_file) as file1:
+                    with open(ctrl2_local_tmp_file) as file2:
+                        return file1.read() == file2.read()
+
     def test_state_failover():
         with dataset("test") as ds:
-            with local_s3_task({
-                "path": f"/mnt/{ds}",
-            }) as task:
+            with local_s3_task({"path": f"/mnt/{ds}"}) as task:
                 call("cloudsync.sync", task["id"], job=True)
-
                 time.sleep(5)  # Job sending is not synchronous, allow it to propagate
 
-                local_job = call("cloudsync.get_instance", task["id"])["job"]
-                local_logs_path = local_job["logs_path"]
-                local_logs = call("filesystem.file_get_contents", local_logs_path)
-
-                remote_job = call("failover.call_remote", "cloudsync.get_instance", [task["id"]])["job"]
-                remote_logs_path = remote_job["logs_path"]
-                remote_logs = call("failover.call_remote", "filesystem.file_get_contents", [remote_logs_path])
-
-                assert local_logs == remote_logs
+                path1 = call("cloudsync.get_instance", task["id"])["job"]["logs_path"]
+                path2 = call("failover.call_remote", "cloudsync.get_instance", [task["id"]])["job"]["logs_path"]
+                assert verify_file_contents_match(path1, path2)
```

Comment:
The issue with this approach is that the test results will not contain the difference between these two strings

Reply:
Why would we want to see the difference between the files in this test? What will that show us?

Reply:
If we assert two things to be equal it's usually useful to see how exactly are they not different, that's what pytest does by default on assertion failure